### DISCUSSION TITLE
How can i use jaeger with nestjs and rabbitMQ?

### DISCUSSION THREAD

**Discussion by amirkangarloo** [jaegertracing/jaeger#3815] [Q&A] [answered]
I want to use Jaeger for tracing microservice app. my app is implemented with Node, nestJS, and RabbitMQ.
When I use Jaeger on one of the services it works, but I can't trace other services because Jaeger needs an HTTP request (https://cloudnative101.dev/electives/dist-trace/#propagating-a-trace-with-injectextract).

How can I trace between services **with Rabbit** and don't use HTTP req?

Thanks.

**Comment by yurishkuro** [comment, marked as answer]
Jaeger is not in charge of producing data out of applications, it is the job of instrumentation, such as opentelemetry. Google search shows many posts about OTEL and Rabbitmq.
